src/engine/Logger.hpp

```diff
@@ -0,0 +1,73 @@
+#pragma once
+
+#include <ctime>
+#include <filesystem>
+#include <fstream>
+#include <iostream>
+
+namespace admirals {
+
+class Logger {
+private:
+    std::ofstream textlog_m;
+    const std::string filename_m = "logFile.txt";
+    const std::time_t startTime_m = std::time(nullptr);
+
+    const std::string timeSinceStart() {
+        const std::time_t currentTime = std::time(nullptr);
+        long long time_taken = currentTime - startTime_m;
+        long sec = time_taken % 60;
+        long min = (time_taken / 60) % 60;
+        long hour = (time_taken / 3600) % 60;
+        std::string return_str = std::to_string(hour) + ":" +
+                                 std::to_string(min) + ":" +
+                                 std::to_string(sec);
+        return return_str;
+    }
+
+    std::string getErrorMessage(std::error_code ec) {
+        switch (ec.value()) {
+        case 1337:
+            return "One code paragraph was too elite.";
+        case 101:
+            return "Engine Error: \n    Value out of bounds";
+        case 102:
+            return "Engine Error: \n    Object not found";
+        case 201:
+            return "Network error:\n    Connection lost";
+        default: // code 0
+            return "unknown error code.";
+        }
+    }
```

Comment:
Add some documentation for these error codes, maybe add enums to make it easier.